contracts/protocol/facets/SequenatialCommitHandlerFacet.sol

```diff
@@ -0,0 +1,211 @@
+        if (_priceDiscovery.direction == Direction.Sell) {
+            require(seller == msgSender(), NOT_VOUCHER_HOLDER);
+        }
+
+        // First call price discovery and get actual price
+        // It might be lower tha submitted for buy orders and higher for sell orders
+        uint256 actualPrice = fulFilOrder(_exchangeId, tokenAddress, _priceDiscovery, _buyer, offer.sellerId);
+
+        // Calculate the amount to be kept in escrow
+        uint256 escrowAmount;
+        {
+            // Get sequential commits for this exchange
+            SequentialCommit[] storage sequentialCommits = protocolEntities().sequentialCommits[_exchangeId];
+
+            {
+                // Calculate fees
+                uint256 protocolFeeAmount = tokenAddress == protocolAddresses().token
+                    ? protocolFees().flatBoson
+                    : (protocolFees().percentage * actualPrice) / 10000;
+
+                // Calculate royalties
+                (, uint256 royaltyAmount) = IBosonVoucher(protocolLookups().cloneAddress[offer.sellerId]).royaltyInfo(
+                    _exchangeId,
+                    actualPrice
+                );
+
+                // Verify that fees and royalties are not higher than the price.
+                require((protocolFeeAmount + royaltyAmount) <= actualPrice, FEE_AMOUNT_TOO_HIGH);
+
+                // Get price paid by current buyer
+                uint256 len = sequentialCommits.length;
+                uint256 currentPrice = len == 0 ? offer.price : sequentialCommits[len - 1].price;
+
+                // Calculate the amount to be immediately released to current voucher owner
+                // escrowAmount =
+                //     actualPrice -
+                //     Math.min(currentPrice, actualPrice - protocolFeeAmount - royaltyAmount);
+                // escrowAmount =
+                //     Math.max(actualPrice-currentPrice,  protocolFeeAmount + royaltyAmount); // can underflow
+                escrowAmount = Math.max(actualPrice, protocolFeeAmount + royaltyAmount + currentPrice) - currentPrice;
+
+                // Update sequential commit
+                sequentialCommits.push(
+                    SequentialCommit({
+                        resellerId: buyerId,
+                        price: actualPrice,
+                        protocolFeeAmount: protocolFeeAmount,
+                        royaltyAmount: royaltyAmount
+                    })
+                );
+            }
+
+            // Make sure enough get escrowed
+            if (_priceDiscovery.direction == Direction.Buy) {
+                if (escrowAmount > 0) {
+                    // Price discovery should send funds to the seller
+                    // Nothing in escrow, need to pull everything from seller
+                    if (tokenAddress == address(0)) {
+                        FundsLib.transferFundsToProtocol(address(weth), seller, escrowAmount);
+                        weth.withdraw(escrowAmount);
```

Comment:
could you please add a comment saying that this is withdrawing from weth to native token to protocol address? 
I had to look into withdraw contract to know what is happening here

Reply:
Added explanation.

Just a heads up - when you will work on normal `commitToOffer` you will be able to get native currency directly from the seller's pool, so no weth shenanigans will be needed :)